pkg/issuer/acme/issue.go

```diff
@@ -158,7 +158,7 @@ func (a *Acme) obtainCertificate(ctx context.Context, crt *v1alpha1.Certificate)
 		return nil, nil, fmt.Errorf("failed to parse returned x509 certificate: %v", err.Error())
 	}
 
-	if a.Context.IssuerOptions.CertificateNeedsRenew(x509Cert) {
+	if a.Context.IssuerOptions.CertificateNeedsRenew(x509Cert, crt.Spec.RenewBefore.Duration) {
```

Comment:
So this whole thing was confusing to me, since there is already a function `scheduleRenewal` called from sync.go which seems to do something similar. I dont really understand why we need both functions. At the very least, I will abstract the logic of getting the defaults into its own function which can be called from both.

Reply:
Sorry, missed your response here.

We have this extra check for validity here, because in some cases, we may have lost the TLS certificate for some reason, however the old one that the ACME server has for us (i.e. the one associated with the order) is *still valid* (without our own definition of valid).

If the certificate from the previous order *is* still valid, then we simply retrieve that cert instead of re-issuing a new certificate altogether. This cuts down on the number of re-validations we need to perform 😄